src/flb_upstream.c

```diff
@@ -841,10 +841,11 @@ int flb_upstream_conn_timeouts(struct mk_list *list)
             }
 
             if (drop == FLB_TRUE) {
-                mk_event_inject(u_conn->evl, &u_conn->event,
-                                MK_EVENT_READ | MK_EVENT_WRITE,
-                                FLB_TRUE);
-
+                if (u_conn->event.status != MK_EVENT_NONE) { /*x*/
+                    mk_event_inject(u_conn->evl, &u_conn->event,
+                                    MK_EVENT_READ | MK_EVENT_WRITE,
+                                    FLB_TRUE);
+                }
                 u_conn->net_error = ETIMEDOUT;
                 prepare_destroy_conn(u_conn);
```

Comment:
Should we set ETIMEDOUT and destroy connection if this is a DNS timeout?

Reply:
I think so because: before inject added, we have u_conn prepared for destroy regardless of if event resumed. https://github.com/fluent/fluent-bit/commit/a9605f3a7698f104ba44e9b804257a191a4bbf76